src/libstd/sys/windows/mod.rs

```diff
@@ -72,10 +72,13 @@ pub fn decode_error_kind(errno: i32) -> ErrorKind {
     }
 }
 
-fn to_utf16_os(s: &OsStr) -> Vec<u16> {
-    let mut v: Vec<_> = s.encode_wide().collect();
-    v.push(0);
-    v
+pub fn to_u16s(s: &OsStr) -> Option<Vec<u16>> {
```

Comment:
I think it'd be fine to have this directly return an `io::Result` (like unix does with `try!` + `CString::new`) so it doesn't need wrappers in sub-modules.